### DISCUSSION TITLE
Translations doesn't work for Text Input fields

### DISCUSSION THREAD

**Discussion by whoiscado** [solspace/craft-freeform#264] [Questions & Answers]
Hey!

`state of the website:` 

German website with an English translation. I have **one** `freeform.php` file in the folder `/translations/en/` with the translation of my German form to English.

The view of the form works fine on the English site.

`problem:`

However, on the German site, the English translation is displayed to me. Except for the checkbox and selection items, all the text input fields have english labels. The checkbox and selection labels are in German like they're written in the form in the freeform backend.

`Possible solution`: I create another `freeform.php` in the `en` folder. But the value to be translated **must be different in some way**. It cannot be the same on both sides, otherwise the complete `freeform.php` file will be ignored. 

How can this be and how can I make it so....

1. only the pages to be translated are displayed in the correct language? (The English translation only on the English page and the German only on the German page).
2. why can't the `freeform.php` contain the same thing on both pages?

Thanks for any help!

**Comment by kjmartens** [comment]
Hi @whoiscado,

Sorry for the trouble you're experiencing with Freeform. I'm just trying to understand what your setup is. In my testing, this works fine to translate field labels and options this way. Can you share what your template code looks like? And are you sure you have the language for that version of the site specified correctly (`site.com/de/` vs `site.com/en` (and setting the correct default language, etc)? 🙂

**Comment by whoiscado** [comment]
Hey @kjmartens,

thanks for your answer. 
```twig
<label class="form-check-label" for="{{ field.idAttribute }}-{{ index }}">
                                    {{ option.label|t|raw }}
                                </label>
```

That's what it looks like, if that's what you mean?!

Yes, everything else works fine. Whether it's the language selection, the craft translations or even the selection fields in the Freeform form.
The only things that do not work are:

- the translations of the headings of the input fields

- create a new freeform.php on the `en` page for example, where then the same sentence is on both pages. If this is the case, the complete `freeform.php` file is skipped. So it's not possible `'ja' => 'ja',` to be written there - To avoid the English translation eg.

Thanks!

**Reply by kjmartens** [reply]
If you're using the Freeform translation file, you should make sure that `{{ option.label|t|raw }}` becomes `{{ option.label|t('freeform')|raw }}`. 

https://docs.solspace.com/craft/freeform/v3/setup/translating.html#front-end-template-translations

Does this resolve the issue for you? 🙂

**Comment by whoiscado** [comment]
Hey @kjmartens,

I changed it, but it didn't fix the error.

Just to clarify:
The main language of the page is German. I actually have only one `freeform.php` file in the `translations/en` folder.
In the form I have the item "Salutation". 
In the German form, it says "Salutation" as the heading and "O Frau O Herr" for the selection fields. 
In the English form it says, "Salutation O Mr O Mrs". So the selection points are translated and the heading remains.

If I now create a `freeform.php` in the `de` folder and write `'Anrede' => 'Anrede'` into it, it remains with "Salutation". But if I write `'Anrede' => 'Anredee'` into it, "Anredee" appears as headline. 

This behavior makes no sense to me and seems to me as if some error already exists there. :D

Thanks a lot for your efforts!
